bench-vortex/benches/compress_benchmark.rs

```diff
@@ -1,50 +1,236 @@
+use std::fs;
+use std::path::Path;
+use std::time::Duration;
+
+use arrow_array::RecordBatch;
 use bench_vortex::data_downloads::BenchmarkDataset;
 use bench_vortex::public_bi_data::BenchmarkDatasets;
 use bench_vortex::public_bi_data::PBIDataset::*;
 use bench_vortex::taxi_data::taxi_data_parquet;
 use bench_vortex::tpch::dbgen::{DBGen, DBGenOptions};
-use bench_vortex::{compress_taxi_data, tpch};
-use criterion::{black_box, criterion_group, criterion_main, Criterion};
-use vortex::{IntoArray, IntoCanonical};
+use bench_vortex::{fetch_taxi_data, tpch};
+use criterion::{
+    black_box, criterion_group, criterion_main, BenchmarkGroup, Criterion, Throughput,
+};
+use parquet::arrow::ArrowWriter;
+use parquet::basic::{Compression, ZstdLevel};
+use parquet::file::properties::WriterProperties;
+use tokio::fs::File;
+use tokio::io::{AsyncSeekExt, AsyncWriteExt};
+use vortex::array::{ChunkedArray, StructArray};
+use vortex::{Array, ArrayDType, IntoArray, IntoCanonical};
+use vortex_dtype::field::Field;
 use vortex_sampling_compressor::compressors::fsst::FSSTCompressor;
 use vortex_sampling_compressor::SamplingCompressor;
+use vortex_serde::layouts::LayoutWriter;
+
+#[derive(serde::Serialize)]
+struct GenericBenchmarkResults<'a> {
+    name: &'a str,
+    value: f64,
+    unit: &'a str,
+    range: f64,
+}
+
+fn ensure_dir_exists(dir: &str) -> std::io::Result<()> {
+    let path = Path::new(dir);
+    if !path.exists() {
+        fs::create_dir_all(path)?;
+    }
+    Ok(())
+}
+
+fn parquet_written_size(array: &Array, filepath: &str, compression: Compression) -> usize {
+    let mut file = std::fs::File::create(Path::new(filepath)).unwrap();
```

Comment:
Just a thought, but the ArrowWriter just needs something that impls `Write`, so instead of writing to file you could just give it a `Vec<u8>` and not worry about pushing random files

Reply:
Man, does everyone come to hate deriving impl's in Rust? /s

Yeah, you're totally right this is silly unnecessary pollution of the filesystem and causes the tests to blow out their disk. I switched to a Cursor<Vec<u8>> which tracks how many bytes have been written.